### PR TITLE
#4173 [Manager] Disable certain action buttons for different Paper element state

### PR DESCRIPTION
**Pull request by konekoya**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #4173)
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [x] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/Pipeline/PipelineApiHelper/connector.js [hunk 1]
```diff
@@ -32,14 +32,11 @@ const connector = () => {
 
   const create = async (params, paperApi) => {
     const { id, name, className } = params;
-    paperApi.disableMenu(id);
```
Comment (konekoya) [inline ohara-manager/client/src/components/Pipeline/PipelineApiHelper/connector.js]: We're no longer exposing this API. Suggested by @wu87988622 👍

File: ohara-manager/client/src/components/Pipeline/Paper/cell/createConnectorCell.js [hunk 2]
```diff
@@ -167,8 +167,11 @@ const createConnectorCell = options => {
         return $buttons.removeClass(cls);
       }
 
+      // Reset before adding new class name
+      $buttons.addClass(cls);
+
       $buttons.each((index, button) => {
-        if (button.className.includes(items)) {
```
Comment (konekoya) [inline ohara-manager/client/src/components/Pipeline/Paper/cell/createConnectorCell.js]: Wrong logic...

File: ohara-manager/client/src/components/Pipeline/Paper/Paper.js [hunk 3]
```diff
@@ -631,6 +697,14 @@ const Paper = React.forwardRef((props, ref) => {
               this.addLink(source.id, target.id);
             }
           });
+
+        // Disable running and failed elements action buttons
+        findElementViews().forEach(elementView =>
```
Comment (konekoya) [inline ohara-manager/client/src/components/Pipeline/Paper/Paper.js]: Once the graph is loaded, we will then update the status accordingly

File: ohara-manager/client/src/components/Pipeline/Paper/Paper.js [hunk 4]
```diff
@@ -281,37 +298,39 @@ const Paper = React.forwardRef((props, ref) => {
...
+      updateStatus(cell);
       // OnChange event handler is called on graph's `add`, `change` and `remove` events
       onChange(paperApi);
+      cellAddRef.current = cell;
     });
 
     graph.on('change', (cell, updates) => {
       if (!_.has(paperApi, 'state.isReady')) return;
       if (_.isEqual(cellChangeRef.current, updates)) return;
-      if (_.has(cell, 'attributes.target.shouldSkipOnChange')) return;
```
Comment (konekoya) [inline ohara-manager/client/src/components/Pipeline/Paper/Paper.js]: Looks like `shouldSkipOnChange` is necessary as we can check the `id` prop in the link

**Comment by konekoya** [comment]
@eechih sorry, I've included two more small fixes in this PR, so need you to review again 😅 

1. Fix Toolbox and PropertyView z-index issue
2. Don't show `loading metrics...` for connectors that are not running yet (@wu87988622's suggestion)

**Comment by eechih** [comment]
> Fix Toolbox and PropertyView z-index issue (reported by you)
Don't show loading metrics... for connectors that are not running yet (@wu87988622's suggestion)

@konekoya These fixes look good~ :)

**Comment by konekoya** [comment]
> @konekoya These fixes look good~ :)

I will backport this to 0.9. Thanks for the help 💯

**Comment by konekoya** [comment]
retry manager-e2e

**Comment by konekoya** [comment]
retry manager-e2e

**Pull request merged**